autofaiss/indices/distributed.py

```diff
@@ -87,25 +99,31 @@ def _add_index(
 
     faiss.omp_set_num_threads(num_cores)
 
-    # load empty trained index
-    empty_index = get_index_from_bytes(broadcast_trained_index_bytes.value)
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        # load empty trained index
+        if isinstance(broadcasted_index_or_path, str):
+            local_index_path = os.path.join(tmp_dir, "index")
+            empty_index = _load_index(broadcasted_index_or_path, local_index_path)
+        else:
+            empty_index = get_index_from_bytes(broadcasted_index_or_path.value)
 
-    batch_size = get_optimal_batch_size(embedding_reader.dimension, memory_available_for_adding)
+        if make_direct_map:
+            initialize_direct_map(empty_index)
```

Comment:
Now that the input index is a string or an index, it becomes necessary to initialize_direct_map when loading it.
We didn't need that before and I feel it was more simple before
Why supporting broadcast_index_or_str? Is it because it could be more efficient in the long term to not broadcast? or do you think it is better to have a more flexible code for your next improvements?

Reply:
As discussed, partitioned indexes will be created in parallel. So I want to avoid broadcasting all indexes to all workers as only a subset of workers will work on an index.
Also, the direct map is not supported when using the add_with_id() method or when merging indexes which is what we use to add embeddings in a distributed way. So what I will do is generating the direct map at the very last step when writing indexing on disk or when returning an index.